mice-and-cats-java/src/main/java/foop/world/World.java

```diff
@@ -65,46 +73,64 @@ public void afterEntityMoved(HashSet<Player> players) {
             return;
         }
 
-        var cat = entities.get(0);
+
+        var cats = entities.stream().filter(entity -> entity.getType() == CAT).toList();
         for (Entity e : entities) {
-            if (cat != e && !e.isDead() && cat.getPosition().equals(e.getPosition()) && cat.isUnderground() == e.isUnderground()) {
+            if (!cats.contains(e)
+                    && cats.stream().map(Entity::isUnderground).anyMatch(c -> c == e.isUnderground())
+                    && !e.isDead()
+                    && cats.stream().map(Entity::getPosition).anyMatch(c -> c.equals(e.getPosition()))) {
```

Comment:
This is wrong because `isUnderground` is basically part of the position. It needs to be something like `entities.stream().filter(entity -> entity.getType() == CAT && !entity.isDead()).anyMatch(cat -> cat.isUnderground() == e.isUnderground() && cat.getPosition().equals(e.getPosition))`